macros/shared/src/main/scala-2/zio/prelude/Macros.scala

```diff
@@ -198,12 +198,12 @@ new _root_.zio.prelude.QuotedAssertion[${c.weakTypeOf[A]}] {
     val (_, _, codeString) = text(f)
     q"""
 new _root_.zio.prelude.QuotedAssertion[${c.weakTypeOf[A]}] {
-  @_root_.zio.prelude.assertionLambdaQuote($f)
+  @_root_.zio.prelude.assertionLambdaQuote(${c.untypecheck(f)})
```

Comment:
Not sure what this `c.untypecheck` thing is doing. 
Comes from suggestion made here: https://github.com/scala-js/scala-js/issues/5127#issuecomment-2641333748